.github/workflows/main.yml

```diff
@@ -12,11 +12,15 @@ jobs:
   build-and-verify:
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
         variant: ['2.5', '3.0']
         java: ['8', '11', '15']
     steps:
       - uses: actions/checkout@v2
+        with:
+          # Codecov needs fetch-depth > 1
+          fetch-depth: 2
```

Comment:
Does it? What happens otherwise?

Reply:
See https://github.com/spockframework/spock/runs/1614531605?check_suite_focus=true

> GitHub Actions detected.
>->  Issue detecting commit SHA. Please run actions/checkout with fetch-depth > 1 or set to 0